test/test_xla_sharding.py

```diff
@@ -116,6 +116,18 @@ def test_inplace_add_with_sharding(self):
     xt.add_(1)  # inplace update should preserve the sharding
     self.assertEqual(sharding_spec, torch_xla._XLAC._get_xla_sharding_spec(xt))
 
+  def test_shard_hashing(self):
+    xt1 = torch.ones(2, 2).to(xm.xla_device())
```

Comment:
There is a function called, `_get_graph_hash` in `_XLAC`. Let's use it to explicitly check (assertions) before and after sharding hashes, and keep the final execution to make sure both the original computation and the new shasrded computation yield the same result.